### PR TITLE
feat: deploy kubebb stack in private cluster

### LINKED ISSUE #59: deploy kubebb stack in private cluster
When kubernetes cluster can not access public network ,we have to use kubebb differently.

## Steps to deploy kubbb stack in a private cluster
1. deploy k8s cluster
2. deploy a private image registry(optional)
3. push all images to private image registry 
 - kubebb/core
 - buildingbase images
 - chartmuseum image
 - ...
4. deploy kubebb core
5. deploy a private component repository
   - use chartmuseum
   - only internal usage
6. push all official components into this private repository
6. create `Repository` into `kubebb/core`
 - with image registry override

## Steps to deploy a component in private cluster
1. push component into private repository
2. push required images into private image registry
3. update `componentplan.yaml` to use private image registry and private component repository
4. apply `componentplan.yaml` and check status

### PR DESCRIPTION
**Pull request by 0xff-dev**
feat: deploy kubebb stack in private cluster
Fix: https://github.com/kubebb/components/issues/59

image

image

### CODE REVIEW

File: docs/core/userguide/privatecluster.md [hunk 1]
```diff
@@ -0,0 +1,501 @@
+---
+sidebar_position: 1
+---
+
+# 从零开始部署内核
```
Comment (bjwswang) [inline docs/core/userguide/privatecluster.md]: title -》 私有集群部署方案

**Review by bjwswang — COMMENTED** [review]
pls place a picture to show the overall catalog in document.

**Comment by 0xff-dev** [comment]
> pls place a picture to show the overall catalog in document.

Done.

File: docs/core/userguide/privatecluster.md [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+
+在使用 kubebb-core 之前我们需要有一个 k8s 集群，版本是 `1.24+`。
+
+## 1. 准备
+
+**因为是在私有环境部署，所以需要通过某个可以联网集群，获取下面所必须的 命令，镜像上传到目标节点。**
+
+### 命令
```
Comment (bjwswang) [inline docs/core/userguide/privatecluster.md]: 命令行工具

File: docs/core/userguide/privatecluster.md [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+### 镜像
+
+- kindest/node:v1.24.4 (创建k8s集群，kind要是用的镜像)
+- kubebb/core:v0.1.2 (kubebb-core 的镜像)
+- alpine:3.15 (用来测试registry是否正常)
+- ghcr.io/helm/chartmuseum:v0.16.0
+
+### 代码仓库
+
+- kubebb/components (这里包含了kubebb的基础chart包, 因为是私有部署，所以需要先clone下来)
```
Comment (bjwswang) [inline docs/core/userguide/privatecluster.md]: add a link

File: docs/core/userguide/privatecluster.md [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+  localRegistryHosting.v1: |
+    host: "localhost:5001"
+    help: "https://kind.sigs.k8s.io/docs/user/local-registry/"
+```
+
+创建集群
+
+```shell
+# 1. create cluster
+kind create cluster --config=conf-with-registry.yamlCreating
```
Comment (bjwswang) [inline docs/core/userguide/privatecluster.md]: wrong config file . Need to remove `Creating`

**Review by bjwswang — APPROVED** [review]

**Pull request merged**